File: changsukuaidi-express-andriod/app/src/main/java/com/changsukuaidi/www/dagger/comp/AppComp.java
package com.changsukuaidi.www.dagger.comp;


import com.changsukuaidi.www.base.BaseApplication;
import com.changsukuaidi.www.dagger.module.ClientModule;
import com.changsukuaidi.www.remote.ClientManager;

import javax.inject.Singleton;

import dagger.Component;

/**
 * @Describe
 * @Author zhouhao
 * @Date 2017/12/21
 * @Contact 605626708@qq.com
 */
@Singleton
@Component(dependencies = BaseComp.class, modules = {ClientModule.class})
public interface AppComp{

    // 网络连接
    ClientManager provideClientManager();

    BaseApplication provideApp();
}

File: WIFIDirect/app/src/main/java/com/thc/www/p2p/comp/AppComp.kt
package com.thc.www.p2p.comp

import com.thc.www.p2p.base.BaseApplication
import com.thc.www.p2p.dao.TokenDao
import com.thc.www.p2p.dao.UserDataBase
import com.thc.www.p2p.module.AppModule
import com.thc.www.p2p.module.DaoModule
import com.thc.www.p2p.module.DataBaseModule
import dagger.Component
import javax.inject.Singleton

/**
 * @Describe
 * @Author thc
 * @Date  2018/4/20
 */
@Singleton
@Component(modules = [AppModule::class, DataBaseModule::class, DaoModule::class])
interface AppComp {
    fun provideApp(): BaseApplication

    // 提供db对象
    fun provideUserDb(): UserDataBase

    // 提供dao对象
    fun provideTokenDao(): TokenDao
}